### ISSUE TITLE
Traffic ports open for ipv4 only (0.0.0.0) in an k8s ipv6-only cluster

### ISSUE THREAD

**Issue by uablrek** [heptio/contour#923]
**What steps did you take and what happened:**

Deploy contour in a k8s ipv6-only cluster with;
```
kubectl apply -f https://j.hept.io/contour-deployment-rbac
```

Configure some simple http back-end and an "ingress". Login to the contour container and examine the listen ports;
```
kubectl -n heptio-contour exec -c contour -it contour-6d78b6fdd9-cb7nt sh
# netstat -lutn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      
tcp        0      0 127.0.0.1:8001          0.0.0.0:*               LISTEN      
tcp        0      0 127.0.0.1:9001          0.0.0.0:*               LISTEN      
tcp        0      0 127.0.0.1:6060          0.0.0.0:*               LISTEN      
tcp        0      0 :::8000                 :::*                    LISTEN      
tcp        0      0 :::8002                 :::*                    LISTEN      
```

Oh, and ipv6 traffic doesn't work of course.

**What did you expect to happen:**

Traffic port should be open with the ipv6 any-address `::` and ipv6 traffic should work.

**Anything else you would like to add:**

The endpoint-fix in #911 must be applied.

With this configuration;
```
      containers:
      - image: gcr.io/heptio-images/contour:master
        imagePullPolicy: Always
        name: contour
        command: ["contour"]
        args:
        - serve
        - --incluster
        - "--envoy-service-http-address=::"
        - "--envoy-service-https-address=::"
```

it works!

I suggest a new manifest for ipv6 is published like;
```
kubectl apply -f https://j.hept.io/contour-deployment-rbac.ipv6
```

with this update and the one in #911 . The file I used is attached.

**Environment:**

- Contour version:
"master"(?) (the image tag is "master")
- Kubernetes version: (use `kubectl version`):
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.0-alpha.0.205+d9f3e96796e2d1", GitCommit:"d9f3e96796e2d154b1d4caa156ba95ff9b01e5b2", GitTreeState:"clean", BuildDate:"2019-02-22T08:07:36Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.0-alpha.0.205+d9f3e96796e2d1", GitCommit:"d9f3e96796e2d154b1d4caa156ba95ff9b01e5b2", GitTreeState:"clean", BuildDate:"2019-02-22T08:07:06Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes installer & version:
None
- Cloud provider or hardware configuration:
None
- OS (e.g. from `/etc/os-release`):
[xcluster](https://github.com/Nordix/xcluster)

[contour-deployment-rbac.ipv6.txt](https://github.com/heptio/contour/files/2935813/contour-deployment-rbac.ipv6.txt)

**Comment by davecheney** [comment]
Is there a way that we can bind to v4 and v6 by default? If so then i'll
make that the default so nobody has to set any flags.

**Comment by uablrek** [comment]
Normally an ipv6 "any" also listen on ipv4 (dual-stack). I tried to use "--stats-address=::" in ipv4 but then the endpoints became not-ready, i.e same as in #911 but reversed. So it does not work out-of-the-box but maybe there is some config to envoy for dual-stack (I don't know envoy). IMO it should be default for "::". The dua-stack can be disabled with `sysctl net.ipv6.bindv6only`. I have not checked if this is done in the contour pod.

**Comment by uablrek** [comment]
In the contour pod;
```
/ # hostname
contour-86559586d6-g9tc8
/ # sysctl net.ipv6.bindv6only
net.ipv6.bindv6only = 0
```

**Comment by uablrek** [comment]
See also; https://github.com/heptio/contour/issues/48#issuecomment-355139324

**Comment by uablrek** [comment]
And here a discussion in envoy; https://github.com/envoyproxy/envoy/issues/351#issuecomment-289490471

**Comment by uablrek** [comment]
The `pv6_v6only=false` config option in the discussion above does not seem to be implemented. The other possibility mentioned is to have different listeners in envoy for ipv4 and ipv6.

**Comment by uablrek** [comment]
Please check the `ipv4_compat` flag in [envoy documentation](https://www.envoyproxy.io/docs/envoy/v1.6.0/api-v2/api/v2/core/address.proto#core-socketaddress).

IMO this should be set to "true" by default if the address is "::" (ipv6-any). Then it should be possible to have the same configuration for ipv4, ipv6 and dual-stack (which is finally under implementation in [k8s](https://github.com/kubernetes/kubernetes/pull/73977))

**Comment by uablrek** [comment]
I have made a (tiny) PR that sets `ipv4_compat` as suggested above. I have verified manually that the same `contour` can be used for ipv6 and ipv4-only k8s clusters if the ipv6-any address is used "::".

**Comment by davecheney** [comment]
Looks like #1162 was merged in 0.13. Closing
